lib/functions.php

```diff
@@ -77,6 +77,13 @@ function connect(string $uri, ClientConnectContext $socketContext = null, Cancel
         } else {
             // Host is not an IP address, so resolve the domain name.
             $records = yield Dns\resolve($host, $socketContext->getDnsTypeRestriction());
+
+            // Usually the faster response should be preferred, but we don't have a reliable way of determining IPv6
+            // support, so we always prefer IPv4 here.
+            \usort($records, function (Dns\Record $a, Dns\Record $b) {
+                return $a === Dns\Record::A ? -1 : 1;
```

Comment:
Meant `$a->getType()` instead of just `$a` I assume?

Reply:
Of course. :wink: